src/schemathesis/specs/openapi/negative.py

```diff
@@ -0,0 +1,115 @@
+        raise NotImplementedError
+
+
+@attr.s(slots=True)
+class MutateRequired(Mutation):
+    def mutate(self, draw: Draw, new_schema: Schema, property_schema: Schema, name: str) -> None:
+        """Make the schema never generate the given property."""
+        required = new_schema.get("required")
+        if required and name in required:
+            required.remove(name)
+        # An optional property still can be generated
+        # To avoid it we need to remove it completely
+        properties = new_schema["properties"]
+        del properties[name]
+
+
+@attr.s(slots=True)
+class MutateType(Mutation):
+    possible_types: Set[str] = attr.ib(
+        factory=lambda: {"string", "integer", "number", "object", "array", "boolean", "null"}
+    )
+
+    def mutate(self, draw: Draw, new_schema: Schema, property_schema: Schema, name: str) -> None:
+        """Change the property type if possible."""
+        to_exclude = set(get_type(property_schema))
+        available_types = self.possible_types - to_exclude
+        if available_types:
+            property_schema["type"] = draw(st.sampled_from(sorted(available_types)))
+
+
+MUTATIONS = {
+    "headers": (MutateRequired(), MutateType()),
+    "cookies": (MutateRequired(), MutateType()),
+    # We can't mutate `required` since it is required by the spec to be valid.
+    # I.e. if any of path parameters is absent, then we can't reach the endpoint
+    "path_parameters": (MutateType(),),
+    "query": (MutateRequired(), MutateType()),
+    "body": (MutateRequired(), MutateType()),
+    "form_data": (MutateRequired(), MutateType()),
+}
+
+
+# Thoughts:
+#   - Consider a request mutated when at least one parameter is mutated. I.e. one from "header", "query", etc.
+#     Some parameters can't be mutated, but it should matter only when nothing at all can't be mutated.
+
+
+def negative_schema(
+    schema: Dict[str, Any], parameter: str, *, custom_formats: Dict[str, st.SearchStrategy[str]]
+) -> st.SearchStrategy:
+    """A strategy for instances, that DO NOT match the input schema."""
+    mutations = MUTATIONS[parameter]
+
+    validator = jsonschema.Draft4Validator(schema)
+    # First we mutate the input schema, so any valid instance for it is invalid to the original schema
+
+    @st.composite  # type: ignore
+    def mutated(draw: Draw) -> Any:
+        new_schema = deepcopy(schema)
+        # Schemathesis creates "object" schemas for all parameter "in" values except "body" (which is taken as is).
+        # In OAS 2 the "type" keyword is a list (in OAS 3 it is a string) therefore we need to convert "type" to a list
+        if "object" in get_type(new_schema):
+            return mutate_object_schema(draw, new_schema, mutations)
+        return {"not": new_schema}
+
+    return (
+        mutated()
+        .flatmap(lambda s: from_schema(s, custom_formats=custom_formats))
+        .filter(lambda i: not validator.is_valid(i))
+    )
```

Comment:
```suggestion
    return mutated().flatmap(
        lambda s: from_schema({"allOf": [{"not": schema}, s]}, custom_formats=custom_formats)
    )
```

In principle `hypothesis-jsonschema` might be smarter about how to canonicalise this; in practice it will often fall back to exactly the same filter implementation in the worst case.